samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/DeliveryBean.java

```diff
@@ -3540,7 +3540,36 @@ private Site getCurrentSite(String id) {
 		}
 		return site;
 	}
-	
+
+	public String getCurrentToolId() {
```

Comment:
I see some other tools (msgcntr, commons) just calling 
ToolManager.getCurrentPlacement().getId();

Does this return the wrong id here?

Reply:
I guess that's what this was doing before right? And getting it wrong being called from another tool . . . Okay

Reply:
I'm surprised a method like this doesn't exist in the SiteService. I wonder if SiteService.siteToolReference(getSiteId(), "sakai.samigo.tool") would return the same?

Reply:
siteToolReference just concatenates what you pass it:

return getAccessPoint(true) + Entity.SEPARATOR + siteId + Entity.SEPARATOR + TOOL_SUBTYPE + Entity.SEPARATOR + toolId;

Problem is you need the placement id (36-char UUID) and not the tool id (sakai.samigo)

Reply:
I see, I think this could still be more of a utility method in SiteService as the only parameters it needs are currentSiteId and the toolId to look for.

If you'd rather avoid a kernel change and think this is only going to be an issue with Samigo and this one feature that's fine with me too.